src/utils/locations.ts

```diff
@@ -61,6 +75,46 @@ export function getMainFunctionTokenLocation(
   return location!;
 }
 
+// As `ReportDescriptor` may contain either `message` or `messageId` prop,
+// we force the presence of `message` property by using the following type alias.
+export type ReportDescriptor = Rule.ReportDescriptor & { message: string };
```

Comment:
Why are we exporting this type?

Reply:
Because it's used in an exported API as parameter